src/app/modules/item/pages/item-current-situation/item-current-situation.component.ts

```diff
@@ -17,52 +15,36 @@ export class ItemCurrentSituationComponent implements OnChanges {
   @Input() itemData?: ItemData;
   @Input() watchedGroup?: Group;
 
-  viewItems?: ViewItem[];
-  viewSelectedIndex = 0;
-  viewSelected?: ViewItem;
+  @ViewChild('historyTab') historyTab?: RouterLinkActive;
+  @ViewChild('chapterTab') chapterTab?: RouterLinkActive;
+  @ViewChild('chapterUserProgressTab') chapterUserProgressTab?: RouterLinkActive;
+
   hideSelection = false;
-  type?: 'Chapter' | 'Task' | 'Course' | 'Skill';
+  showChapterUserProgress = false;
   logColumns?: Column[];
+  logItemCaption?: string;
 
   constructor() {}
 
   ngOnChanges(): void {
-    if (!this.itemData) {
+    const type = this.itemData?.item?.type;
+
+    if (!type) {
       return;
     }
 
-    this.type = this.itemData.item.type;
-    this.viewItems = this.getViewItems();
-    this.hideSelection = this.getHideSelection();
-    this.logColumns = this.getLogColumns();
-    this.changeView(this.viewSelectedIndex);
-  }
-
-  getViewItems(): ViewItem[] {
-    const logItemCaption = !this.watchedGroup && this.type === 'Chapter'
-      || !!this.watchedGroup && !!this.type && [ 'Task', 'Course' ].includes(this.type) ? $localize`History` : $localize`Log view`;
-    const chapterViewValue = !this.watchedGroup && this.type === 'Chapter' ? 'chapter-user-progress' : 'chapter';
-
-    return [
-      { label: logItemCaption, value: 'log' },
-      { label: $localize`Chapter view`, value: chapterViewValue },
-    ];
-  }
-
-  onViewChanged(selectedIdx: number): void {
-    this.changeView(selectedIdx);
-  }
+    this.hideSelection = this.getHideSelection(type);
+    this.logColumns = this.getLogColumns(type);
```

Comment:
I don't think it should be the responsibility of this component to decide the columns of its subcomponent.

Reply:
Fixed